packages/common/http/src/xhr.ts

```diff
@@ -70,7 +77,8 @@ interface PartialResponse {
  */
 @Injectable()
 export class HttpXhrBackend implements HttpBackend {
-  constructor(private xhrFactory: XhrFactory) {}
+  constructor(private xhrFactory: XhrFactory, @Inject(JSON_PARSER) private jsonParser: JsonParser) {
```

Comment:
How about making this optional, then we wouldn't have to define a provider for the default case:

```suggestion
  constructor(private xhrFactory: XhrFactory, @Optional() @Inject(JSON_PARSER) private jsonParser?: JsonParser) {
  if (jsonParser == null) { // note == matches both `null` and `undefined`
    this.jsonParser = JSON;
  }
```